packages/core/cli.js

```diff
@@ -42,22 +42,40 @@ listeners.forEach(listener => process.removeListener("warning", listener));
 
 const inputStrings = process.argv.slice(2);
 
-const userWantsGeneralHelp =
-  inputStrings.length === 0 ||
-  (inputStrings.length === 1 && ["help", "--help"].includes(inputStrings[0]));
-
-if (userWantsGeneralHelp) {
-  const { displayGeneralHelp } = require("./lib/command-utils");
-  displayGeneralHelp();
-  process.exit(0);
-}
-
 const {
   getCommand,
   prepareOptions,
-  runCommand
+  runCommand,
+  displayGeneralHelp
 } = require("./lib/command-utils");
 
+//User only enter truffle with no commands, let's show them what's available.
+if (inputStrings.length === 0) {
+  displayGeneralHelp();
+  process.exit();
+}
+
+//if `help` or `--help` is in the command, validate and transform the input argument for help
+if (
+  inputStrings.some(inputString => ["help", "--help"].includes(inputString))
+) {
+  //when user wants general help
+  if (inputStrings.length === 1) {
+    displayGeneralHelp();
+    process.exit();
+  }
+
+  //check where is --help used, mutate argument into a proper help command
+  const helpIndex = inputStrings.indexOf("--help");
```

Comment:
I'm not going to say this is necessary but I think a better array method to use would have been `inputStrings.includes("--help")` as it is just easier to understand.